Objects/codeobject.c

```diff
@@ -318,6 +318,11 @@ init_code(PyCodeObject *co, struct _PyCodeConstructor *con)
         co->co_nfreevars = 0;
     }
 
+    if(!is_new) {
+        Py_XDECREF(co->co_filename);
+        Py_XDECREF(co->co_name);
+        Py_XDECREF(co->co_qualname);
+    }
```

Comment:
These pointers can be garbage in a new object. I am looking at whether this can be fixed in the argument clinic (init them to NULL).

Reply:
Or you could zero these out in `_PyCode_New()`?